### PR TITLE
Moves to Maven wrapper to stabilize the version in use

### PR DESCRIPTION
**Pull request by adriancole**
- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

### New feature or improvement
This switches usage from `mvn` to `./mvnw` (the maven wrapper). This will ensure those building will have similar experience due to having a fixed maven version.

### CODE REVIEW

**Comment by wu-sheng** [comment]
Considering this PR includes source codes, let us figure about the License things.

**Comment by wu-sheng** [comment]
@michaelsembwever @WillemJiang Do you have experience about using `gradlew` or `mvnw` in Apache repository?

**Comment by wu-sheng** [comment]
Maven wrapper(shell and cmd) is already under ASF Licensed.

- https://github.com/bdemers/maven-wrapper
- https://github.com/takari/maven-wrapper/

**Comment by ascrutae** [comment]
There has some configuration item which install [ojdbc jar](https://github.com/adriancole/incubator-skywalking/blob/mvn-wrapper/pom.xml#L450) and [resin jar](https://github.com/adriancole/incubator-skywalking/blob/mvn-wrapper/pom.xml#L450) in `pom.xml`, Is there execute command need to change to `./mvnw` command?

BTW, I found that I execute `./mvnw install` command  with error message and execute `./mvnw install -Dmaven.test.skip` without error message in my local environment, Is there same problem in your environment, if not, I will recheck my environment. Here is the error logs:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.20.1:test (default-test) on project apm-util: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.20.1:test failed.: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.20.1:test (default-test) on project apm-util: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.20.1:test failed.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
....
Caused by: java.lang.NullPointerException
    at org.apache.maven.surefire.shade.org.apache.commons.lang3.SystemUtils.isJavaVersionAtLeast (SystemUtils.java:1626)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.getEffectiveJvm (AbstractSurefireMojo.java:2107)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.getForkConfiguration (AbstractSurefireMojo.java:1976)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider (AbstractSurefireMojo.java:1111)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked (AbstractSurefireMojo.java:954)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute (AbstractSurefireMojo.java:832)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
```

**Comment by adriancole** [comment]
it might be helpful for me to add version updates as a lot of the plugins
are a bit old. probably you will have problems with new jdk. shall I add
updates?

**Comment by adriancole** [comment]
summary is that updating the versions here is a good thing because the maven version bundled has newer versions than what the pom currently pins. Best way to have smooth is to have versions set to same or more recent version.. might not be behind the error, but easy way to rule out problems

**Comment by wu-sheng** [comment]
I mean, if update jdk requirement, we need to change compile and install document. Update maven plugin version, should be fine.

Could you help on the update?

**Comment by adriancole** [comment]
This pull request just moves to maven wrapper (and bumps common plugin versions). I setup maven enforcer to fail fast on JDK > 1.8 because other work needs to happen to support later versions.

The task of stabilizing maven is different than the task of using latest java, even if this task will help make using recent java easier. Make sense?

**Comment by adriancole** [comment]
PS there are other plugins that need refactoring.. I noticed some copy paste around things like grpc. I don't have time to investigate each of these, but at least getting the standard plugins in shape is hopefully helpful.

File: apm-collector/apm-collector-remote/collector-remote-grpc-provider/pom.xml [hunk 1]
```diff
@@ -55,14 +55,6 @@
             </extension>
         </extensions>
         <plugins>
-            <plugin>
```
Comment (adriancole) [inline apm-collector/apm-collector-remote/collector-remote-grpc-provider/pom.xml]: this plugin is setup in the root pom and inherited, this is why I deleted the many times it was copy pasted

File: apm-application-toolkit/apm-toolkit-log4j-2.x/pom.xml [hunk 2]
```diff
@@ -35,27 +35,4 @@
...
         </dependency>
     </dependencies>
-
-
-    <build>
-        <plugins>
-            <plugin>
-                <!-- 源码插件 -->
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-source-plugin</artifactId>
```
Comment (adriancole) [inline apm-application-toolkit/apm-toolkit-log4j-2.x/pom.xml]: this is also in the main pom, so no need to copy/paste in every module

**Comment by adriancole** [comment]
net down 450+ lines :) I think we paid for the shell scripts heh

**Comment by adriancole** [comment]
FYI I mentioned on gitter, but I'm not planning to do more work on this PR. Some other day I might help with JDK 10, but I have other things to work on today. Cheers!

**Comment by adriancole** [comment]
just rebased as I noticed master moved. this invalidated prior approvals @wu-sheng @michaelsembwever

**Comment by adriancole** [comment]
just rebased as I noticed master moved. this invalidated prior approvals @wu-sheng @michaelsembwever

**Pull request merged**